include/aikido/control/TrajectoryExecutor.hpp

```diff
@@ -29,7 +29,8 @@ class TrajectoryExecutor
   /// \param _traj Trajectory to be executed.
   /// \return future<void> for trajectory execution. If trajectory terminates
   ///        before completion, future will be set to a runtime_error.
-  virtual std::future<void> execute(trajectory::TrajectoryPtr _traj) = 0;
+  virtual std::future<void> execute(const trajectory::ConstTrajectoryPtr& traj)
```

Comment:
Nit: Could we change this to const raw pointer so that shared pointer is not required to call this function? In some concrete implementation requires the ownership?